components/match2/commons/starcraft_starcraft2/match_group_input_starcraft_ffa.lua

```diff
@@ -6,547 +6,430 @@
+
+	local games = MatchFunctions.extractMaps(match, opponents)
+
+	local finishedInput = match.finished --[[@as string?]]
+	match.bestof = tonumber(match.firstto) or tonumber(match.bestof)
+
+	match.finished = MatchFunctions.isFinished(match, opponents)
+	match.mode = MODE_FFA
+
+	if MatchGroupInputUtil.isNotPlayed(match.winner, finishedInput) then
+		match.finished = true
+		match.status = 'notplayed' -- according to RFC ;)
+		match.resulttype = MatchGroupInputUtil.RESULT_TYPE.NOT_PLAYED
+		match.extradata = {ffa = 'true'}
+		return match
 	end
 
-	--apply vodgames
-	for index = 1, MAX_NUM_VODGAMES do
-		local vodgame = match['vodgame' .. index]
-		if Logic.isNotEmpty(vodgame) and Logic.isNotEmpty(match['map' .. index]) then
-			match['map' .. index].vod = match['map' .. index].vod or vodgame
-		end
+	match.pbg = MatchFunctions.getPBG(match)
+
+	local autoScoreFunction = MatchGroupInputUtil.canUseAutoScore(match, games)
+		and not Logic.readBool(match.noscore)
+		and MatchFunctions.calculateMatchScore(games, opponents)
+		or nil
+
+	Array.forEach(opponents, function(opponent, opponentIndex)
+		opponent.score, opponent.status = MatchGroupInputUtil.computeOpponentScore({
+			walkover = match.walkover,
+			winner = match.winner,
+			opponentIndex = opponentIndex,
+			score = opponent.score,
+		}, autoScoreFunction)
+	end)
+
+	Array.forEach(opponents, function(opponent)
+		opponent.placement = tonumber(opponent.placement)
+	end)
+
+	if match.finished then
+		match.resulttype = MatchGroupInputUtil.getResultType(match.winner, finishedInput, opponents)
+		match.walkover = MatchGroupInputUtil.getWalkover(match.resulttype, opponents)
+		StarcraftFfaMatchGroupInput._setPlacements(opponents)
+		match.winner = StarcraftFfaMatchGroupInput._getWinner(opponents, match.winner, match.resulttype)
 	end
 
-	match = StarcraftFfaInput._matchWinnerProcessing(match, numberOfOpponents, noscore)
+	Array.forEach(opponents, function(opponent)
+		opponent.extradata = opponent.extradata or {}
+		opponent.extradata.noscore = Logic.readBool(match.noscore)
+		opponent.extradata.bg = MatchFunctions.readBg(opponent.bg)
+			or match.pbg[opponent.placement]
+			or DEFUALT_BACKGROUND
+
+		-- todo: get rid of the damn alias ...
+		if Logic.isEmpty(opponent.advances) and Logic.isNotEmpty(opponent.win) then
+			opponent.advances = opponent.win
+			mw.ext.TeamLiquidIntegration.add_category('Pages with ffa matches using `win=` in opponents')
+		end
+
+		opponent.extradata.advances = Logic.readBool(opponent.advances)
+			or (match.bestof and (opponent.score or 0) >= match.bestof)
+			or opponent.extradata.bg == ADVANCE_BACKGROUND
+			or opponent.placement == 1
+	end)
+
+	match.opponents = opponents
+	match.games = games
+
+	match.extradata = MatchFunctions.getExtraData(match)
 
 	return match
 end
+---@param match table
+---@param opponents {score: integer?}[]
+---@return boolean
+function MatchFunctions.isFinished(match, opponents)
```

Comment:
Commons can't be used why?

Reply:
because several parts of it are not applicable and would cause errors for the ffa shit
- `hasSpecialStatus` would return true even if only one of the opponents has such a status
- here a match can be finished if it has no scores (hence `not MatchGroupInputUtil.hasScore(opponents)` returning ``false` is wrong for sc(2) ffa)
- `majorityHasBeenWon` is not what i do here i go by placements instead as sc(2) ffa can have placements without scores